src/app/shared/helpers/link.helpers.ts

```
@@ -15,3 +18,37 @@ export function validateExternalReferenceUrl(externalRef?: string): boolean {
     return regexp.test(externalRef.toLowerCase());
   }
 }
+
+export function getDetailsPageLink(itemUuid?: string, itemType?: RegistrationEntityTypes, subpagePath?: string) {
+  const isValid = itemUuid != undefined && itemType != undefined;
+  if (isValid) {
+    switch (itemType) {
+      case 'data-processing-registration':
+        return getDetailsPagePath(AppPath.dataProcessing, itemUuid, subpagePath);
+      case 'it-contract':
+        return getDetailsPagePath(AppPath.itContracts, itemUuid, subpagePath);
+      case 'it-interface':
+        return getDetailsPagePath(`${AppPath.itSystems}/${AppPath.itInterfaces}`, itemUuid, subpagePath);
+      case 'it-system':
+        return getDetailsPagePath(`${AppPath.itSystems}/${AppPath.itSystemCatalog}`, itemUuid, subpagePath);
+      case 'it-system-usage':
+        return getDetailsPagePath(`${AppPath.itSystems}/${AppPath.itSystemUsages}`, itemUuid, subpagePath);
+      case 'organization':
+        return getDetailsPagePath(`${AppPath.organization}/${AppPath.structure}`, itemUuid, subpagePath);
+      default:
+        console.error('Unmapped link itemType', itemType);
+        return undefined;
+    }
+  } else {
+    console.error('Details page link incorrectly configured. Got (uuid,type)', itemUuid, itemType);
+    return undefined;
```

Should it just throw an error instead of returning undefined? This looks like it's failing silently.